mlrun/artifacts/base.py

```diff
@@ -26,7 +27,739 @@
 calc_hash = True
 
 
+class ArtifactMetadata(ModelObj):
+    _dict_fields = ["key", "project", "iter", "tree", "description", "hash"]
+    _extra_fields = ["updated", "labels"]
+
+    def __init__(
+        self, key=None, project=None, iter=None, tree=None, description=None, hash=None
+    ):
+        self.key = key
+        self.project = project
+        self.iter = iter
+        self.tree = tree
+        self.description = description
+        self.hash = hash
+        self.labels = {}
+        self.updated = None
+
+    def base_dict(self):
+        return super().to_dict()
```

Comment:
What's the idea of the extra fields and the base dict ?

Reply:
This is a pre-existing mechanism for artifacts, where the `base` only extract part of the fields and is used in some cases, while the "regular" `to_dict` does the full extraction. I maintained this distinction for backwards compatibility.